BEP7.md

```diff
@@ -0,0 +1,268 @@
+
+**Data Structure** **for Freeze Operation**: A data structure is needed to represent the freeze operation
+
+| **Field** | **Type** | **Description**                                                 |
+| :--------- | :-------- | :------------------------------------------------------------ |
+| Denom     | string   | The symbol of a collection - e.g. NNB-B90                       |
+| Id        | string   | The id of a token in the collection                             |
+| Amount    | int64    | The unfreeze amount can have a maximum of 8 digits of decimal, and the value is boosted by 1e8 to store in an int64. This amount should be no less than the frozen amount |
+
+**Freeze Process:**
+
+- Address-holder signed a freeze transaction and make it broadcasted to one of Binance Chain nodes
+- The Binance Chain node will check this transaction. If there is no error, then this transaction will be broadcasted to other Binance Chain nodes
+- Freeze transaction is committed on the blockchain by block proposer
+- Validators will verify the transaction initiator’s balance is no less than the frozen amount. The fee will be deducted from the transaction initiator’s address.  
+- This amount of tokens in the address of the transaction initiator will be moved from balance to frozen.
+
+#### 6.2.4  Unfreeze Tokens
+
+Unfreezing is to unlock some of the frozen tokens in the user's account and make them liquid again.
+
+**Data Structure** **for Unfreeze Operation**: A data structure is needed to represent the freeze/unfreeze operation
+
+| **Field** | **Type** | **Description**                                              |
+| :--------- | :-------- | :------------------------------------------------------------ |
+| Denom     | string   | The symbol of a collection - e.g. NNB-B90                       |
+| Id        | string   | The id of a token in the collection                             |
+| Amount    | int64    | The unfreeze amount can have a maximum of 8 digits of decimal, and the value is boosted by 1e8 to store in an int64. This amount should be no less than the frozen amount |
+
+**Unfreeze Process:**
+
+- Address-holder signed an unfreeze transaction and make it broadcasted to one of Binance Chain nodes
+- The Binance Chain node will check this transaction. If there is no error, then this transaction will be broadcasted to other Binance Chain nodes
+- Unfreeze transaction is committed on the blockchain by block proposer
+- Validators will verify the transaction initiator’s frozen balance is no less than the required amount. The fee will be deducted from the address of the transaction source.
+- This amount of token will be moved from frozen to balance in the transaction initiator’s address.
+
```

Comment:
should we add **burn token** to BEP7-NFT token?

Reply:
We should include the Burn Token so as to implement across chain communication compatibility for these use cases here are creative examples of needing to invoke a Burn Function:
 
"User's could reverse the process by sending their ERC-721 Virtual Cards to the Wallet address which would then **burn** the tokens and issue an equivalent number of in game items to the user in the game" -Virtual In-game ATM... Tokeninze Second Life
https://blog.0xproject.com/the-inaugural-0x-team-hackathon-8359b68b98e7

Decentralized asset mint (color) and **Burn** (uncolor) functions asset exchange BCH using OP_RETURN -Open Assets SPV Wallets use EPOBC e.exchange color kernels (ColorCoin) BitCoin Cash.
https://www.yours.org/content/colored-coins-in-bitcoin-cash-b26804e05964/
https://news.bitcoin.com/developers-invoke-the-idea-of-bitcoin-cash-based-colored-coins/